packages/dapp/src/components/listing/CommitVoteDetail.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from "react";
+import TransactionButton from "../utility/TransactionButton";
+import { InputElement, StyledFormContainer, FormValidationMessage, FormGroup } from "../utility/FormElements";
+import { TwoStepEthTransaction } from "@joincivil/core";
+import { commitVote, requestVotingRights } from "../../apis/civilTCR";
+import BigNumber from "bignumber.js";
+
+export interface CommitVoteDetailProps {
+  challengeID: BigNumber;
+}
+
+export interface CommitVoteDetailState {
+  isVoteTokenAmtValid: boolean;
+  voteOption?: string;
+  salt?: string;
+  numTokens?: string;
+}
+
+class CommitVoteDetail extends React.Component<CommitVoteDetailProps, CommitVoteDetailState> {
+  constructor(props: any) {
+    super(props);
+
+    this.state = {
+      isVoteTokenAmtValid: false,
+    };
+  }
+
+  public render(): JSX.Element {
+    return (
+      <StyledFormContainer>
+        <h3>Vote On Challenge</h3>
+
+        {/* @TODO(jon): Change this to a hidden form field.
```

Comment:
in the dapp there's no way that you'd ever submit formdata so there's no reason to store extra data in the DOM.  If the user doesn't need to see it, you can just keep it in state or props or redux when we get that setup.

Reply:
Doh! Very good point. I'll change the comment but leave the field in now for debugging's sake.